components/engine_traits/src/tablet.rs

```diff
@@ -69,20 +71,67 @@ impl<EK: Clone> CachedTablet<EK> {
     }
 }
 
+/// Context to be passed to `TabletFactory`.
+#[derive(Clone)]
+pub struct TabletContext {
+    /// ID of the tablet. It is usually the region ID.
+    pub id: u64,
+    /// Suffix the tablet. It is usually the index that the tablet starts accept
+    /// incremental modification. The reason to have suffix is that we can keep
+    /// more than one tablet for a region.
+    pub suffix: Option<u64>,
+    /// The expected start key of the tablet. The key should be in the format
+    /// tablet is actually stored, for example should have `z` prefix.
+    ///
+    /// Any key that is smaller than this key can be considered obsolete.
+    pub start_key: Box<[u8]>,
+    /// The expected end key of the tablet. The key should be in the format
+    /// tablet is actually stored, for example should have `z` prefix.
+    ///
+    /// Any key that is larger than or equal to this key can be considered
```

Comment:
I think start_key and end_key can be optional instead. In some cases these are not needed (such as destroy_tablet)

Reply:
It depends. For the moment it doesn't need it. But if we try to cleanup the data using rocksdb's API (DB::destroy), we will need options, which may need start key and end key.